Replay search: Searching by replay ID with "contains" condition throws API error

**Describe the bug**
If you search for replays by ID, and use the "contains" modifier, creating a query like this:

```
id=="*10814464*"
```

You receive an error message:

```
java.util.concurrent.CompletionException: org.springframework.web.client.HttpClientErrorException$BadRequest: 400 Bad Request
	at org.springframework.aop.interceptor.AsyncExecutionAspectSupport.lambda$doSubmit$3(AsyncExecutionAspectSupport.java:279)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.springframework.web.client.HttpClientErrorException$BadRequest: 400 Bad Request
	at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:79)
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:122)
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:102)
	at com.faforever.client.api.JsonApiErrorHandler.handleError(JsonApiErrorHandler.java:31)
	at org.springframework.security.oauth2.client.http.OAuth2ErrorHandler.handleError(OAuth2ErrorHandler.java:172)
	at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)
	at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:778)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:736)
	at org.springframework.security.oauth2.client.OAuth2RestTemplate.doExecute(OAuth2RestTemplate.java:128)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:670)
	at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:311)
	at com.faforever.client.api.FafApiAccessorImpl.getPage(FafApiAccessorImpl.java:630)
	at com.faforever.client.api.FafApiAccessorImpl.getPage(FafApiAccessorImpl.java:617)
	at com.faforever.client.api.FafApiAccessorImpl.findReplaysByQuery(FafApiAccessorImpl.java:335)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:205)
	at com.sun.proxy.$Proxy70.findReplaysByQuery(Unknown Source)
	at com.faforever.client.remote.FafService.findReplaysByQuery(FafService.java:290)
	at com.faforever.client.remote.FafService$$FastClassBySpringCGLIB$$6c91c77e.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:749)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
	at org.springframework.aop.interceptor.AsyncExecutionAspectSupport.lambda$doSubmit$3(AsyncExecutionAspectSupport.java:276)
	... 4 more
```

**To Reproduce**
Issue this query: `id=="*10814464*"`

**Expected behavior**
I would expect "contains" to not appear as an option for Replay ID search, since it isn't valid. Alternately, it would work.

This bug looks related to https://github.com/FAForever/downlords-faf-client/issues/1097, and may or may not be fixed by a fix of that bug.

**OS**
Windows

--- a/src/main/java/com/faforever/client/query/SearchablePropertyMappings.java
+++ b/src/main/java/com/faforever/client/query/SearchablePropertyMappings.java
@@ -59,6 +59,7 @@ public class SearchablePropertyMappings {
       .put("latestVersion.height", new Property("map.height", true))
       .put("latestVersion.folderName", new Property("map.folderName", false))
       .put("latestVersion.ranked", new Property("map.ranked", true))
+      .put("latestVersion.id", new Property("map.versionId", false))
 
       .build();
 

--- a/src/main/java/com/faforever/client/query/SpecificationController.java
+++ b/src/main/java/com/faforever/client/query/SpecificationController.java
@@ -89,6 +89,7 @@ public class SpecificationController implements Controller<Node> {
           .put(Enum.class, Arrays.asList(EQ, NE, IN, NIN))
           .put(ComparableVersion.class, Arrays.asList(EQ, NE, GT, GTE, LT, LTE, IN, NIN))
           .build();
+  public static final String ID_PROPERTY_NAME = "id";
 
   private final I18n i18n;
   private final FilteredList<ComparisonOperator> comparisonOperators;
@@ -178,6 +179,11 @@ public class SpecificationController implements Controller<Node> {
   }
 
   private boolean isOperatorApplicable(ComparisonOperator comparisonOperator, String propertyName) {
+    if (propertyName.endsWith(String.format(".%s", ID_PROPERTY_NAME)) || propertyName.equals(ID_PROPERTY_NAME)) {
+      //Because the dto's id property is String but is internally mapped to an int we need to do a small hack here
+      return VALID_OPERATORS.get(Number.class).contains(comparisonOperator);
+    }
+
     Class<?> propertyClass = ClassUtils.resolvePrimitiveIfNecessary(getPropertyClass(propertyName));
 
     for (Class<?> aClass : VALID_OPERATORS.keySet()) {
@@ -282,6 +288,11 @@ public class SpecificationController implements Controller<Node> {
   }
 
   private Property getProperty(QBuilder<?> qBuilder, String property, Class<?> fieldType) {
+    if (property.endsWith(String.format(".%s", ID_PROPERTY_NAME)) || property.equals(ID_PROPERTY_NAME)) {
+      //Because the dto's id property is String but is internally mapped to an int we need to do a small hack here
+      return qBuilder.intNum(property);
+    }
+
     Property prop;
     if (ClassUtils.isAssignable(Number.class, fieldType)) {
       prop = qBuilder.intNum(property);

--- a/src/main/resources/i18n/messages.properties
+++ b/src/main/resources/i18n/messages.properties
@@ -376,6 +376,7 @@ map.size=Size
 map.rating=Rating
 map.players=Players
 map.version=Version
+map.versionId=Map version ID
 map.versionFormat=v{0}
 map.folderName=Folder name
 map.ranked=Ranked

